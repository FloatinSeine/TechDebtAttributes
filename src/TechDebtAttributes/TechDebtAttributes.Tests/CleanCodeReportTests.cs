using System.Reflection;
using ApprovalTests;
using ApprovalTests.Reporters;
using ExampleUsage;
using Xunit;

namespace TechDebtAttributes.Tests
{
    public class CleanCodeReportTests
    {
        [Fact]
        [UseReporter(typeof(NotepadLauncher))]
        public void ShouldRenderReport()
        {
            var assemblyToReportOn = Assembly.GetAssembly(typeof(SomeThingToClean));

            Approvals.Verify(CleanCodeReporter.GenerateReport(assemblyToReportOn));
        }
    }
}
