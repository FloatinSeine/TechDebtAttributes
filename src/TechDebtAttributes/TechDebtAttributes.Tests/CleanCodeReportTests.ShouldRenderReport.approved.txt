***Start of Tech Debt Report - finding all [TechDebt] attribute usages

Type	Beneft	Severity	Effort	Code	Description	Reviewer	BacklogId
Yagni	0.7	Mild	Medium	ExampleUsage.SomeThingToClean.Void SomeMethod2()	Pointless method you wont need.		
Solid	1.5	Painful	Minor	ExampleUsage.SomeThingToClean.Void SomeMethod()	Breaks OCP		
Solid	0.7	Mild	Medium	ExampleUsage.SomeThingToClean	Lacks an interface.	Steve	1000

***End.
